### PR TITLE
Introduction of a way to provide a configuration file and folder from the CLI.

### LINKED ISSUE #377: BUG: PyF in virtualvenv fails loading conf files
## Description

If you follows https://pyfunceble.readthedocs.io/en/dev/installation/index.html#for-development

then Pyfunceble completely fails to use the files from `~/.config/PyFunceble/`

## Configuration

seems like none

## Reproduction

1. follow https://pyfunceble.readthedocs.io/en/dev/installation/index.html#for-development
2. pyfunceble -f `anyTestFile`

## Expected behavior

Using the usual path to search for config files, in this case `~/.config/PyFunceble/`

## Screenshots

image

image

image

image

## Versions

**OS:** Ubuntu 24

**Python Version:** 3.12

**PyFunceble Version:** pyfunceble 4.2.26.dev (Blue Duckling: Ixora)

## Additional context

Add CLI option to force a conf dir?

### PR DESCRIPTION
**Pull request by funilrys**
This patch fixes #377.

Contributors:
  * @spirillen

---
Hey @spirillen & others. I'm lookinf for tester before getting this merged. Could you please test this branch and let me know ?

Thanks!

### CODE REVIEW

File: PyFunceble/cli/entry_points/pyfunceble/cli.py [hunk 1]
```diff
@@ -1197,6 +1197,18 @@ def get_default_group_data() -> List[Tuple[List[str], dict]]:
                 "version": "%(prog)s " + PyFunceble.storage.PROJECT_VERSION,
             },
         ),
+        (
+            [
+                "--config-file",
```
Comment (spirillen) [inline PyFunceble/cli/entry_points/pyfunceble/cli.py:1202]: `--config-file` => `--config-dir`

**Comment by spirillen** [comment]
As mentioned in the comment, https://github.com/funilrys/PyFunceble/pull/380/files#r1771238700, then none of the config/setup files from the default root config dir are picked up when running in python venv as it is a virtual closed environment. This means it is the dir_path to the config files "collection" not only `.PyFunceble.overwrite.yaml`

```bash
/home/$USER/.config/PyFunceble
├── /home/$USER/.config/PyFunceble/dir_structure.json
├── /home/$USER/.config/PyFunceble/iana-domains-db.json
├── /home/$USER/.config/PyFunceble/.idea
│   ├── /home/$USER/.config/PyFunceble/.idea/.gitignore
│   ├── /home/$USER/.config/PyFunceble/.idea/misc.xml
│   ├── /home/$USER/.config/PyFunceble/.idea/modules.xml
│   ├── /home/$USER/.config/PyFunceble/.idea/PyFunceble.iml
│   └── /home/$USER/.config/PyFunceble/.idea/workspace.xml
├── /home/$USER/.config/PyFunceble/inactive.csv
├── /home/$USER/.config/PyFunceble/ipv4_reputation.data
├── /home/$USER/.config/PyFunceble/public-suffix.json
├── /home/$USER/.config/PyFunceble/.pyfunceble-env
├── /home/$USER/.config/PyFunceble/.pyfunceble_intern_downtime.json
├── /home/$USER/.config/PyFunceble/.PyFunceble.overwrite.yaml
├── /home/$USER/.config/PyFunceble/.PyFunceble.yaml
├── /home/$USER/.config/PyFunceble/user_agents.json
└── /home/$USER/.config/PyFunceble/whois.csv
```

**Comment by funilrys** [comment]
@spirillen  added a `--config-dir` option (also). Please test.

**Comment by spirillen** [comment]
> @spirillen added a `--config-dir` option (also). Please test.

I will, is it released on pipy or repo only?

**Pull request merged**